python/ray/serve/_private/deployment_state.py

```diff
@@ -1961,9 +2049,13 @@ def _check_and_update_replicas(self, target_num_replicas: int):
                 # enters the "UNHEALTHY" status until the replica is
                 # recovered or a new deploy happens.
                 if replica.version == self._target_state.version:
-                    self._curr_status_info: DeploymentStatusInfo = DeploymentStatusInfo(
-                        name=self.deployment_name,
+                    # If the deployment transitioned from HEALTHY to
+                    # UNHEALTHY, then there isn't an explicit status
+                    # trigger since it wasn't because a new deployment or
+                    # an update failed.
```

Comment:
This comment is misleading -- failing a health-check is an explicit status trigger

Reply:
Outdated comment, updated!